packages/discord.js/src/structures/InteractionCollector.js

```diff
@@ -150,7 +150,13 @@ class InteractionCollector extends Collector {
     if (this.interactionType && interaction.type !== this.interactionType) return null;
     if (this.componentType && interaction.componentType !== this.componentType) return null;
     if (this.messageId && interaction.message?.id !== this.messageId) return null;
-    if (this.messageInteractionId && interaction.message?.interaction?.id !== this.messageInteractionId) return null;
+    if (
+      this.messageInteractionId &&
+      interaction.message?.interaction?.id &&
```

Comment:
Is this needed?

Reply:
You're right, this isn't needed because if they're both falsey, the below !== comparison will return true anways. Removed.

Reply:
I'm not sure why I removed this looking back. They can't both be falsey when the `this.messageInteractionId` check is there. If `interaction.message?.interaction?.id` is falsey, then they're not equal to each other, this will evaluate to true, and it will return null.